src/app/components/order_input/AmountInput.tsx

```diff
@@ -78,18 +79,20 @@ export function AmountInput(props: TokenInputFiledProps) {
           <span>{symbol}</span>
         </div>
 
-        <input
+        <IMaskInput
           disabled={disabled || false}
-          type="number"
+          mask={Number}
+          scale={targetToken.decimals}
+          radix="."
```

Comment:
the dot here is to tell IMaskInput about the decimal separator? It might be different depending on user settings, so it probably won't work if the separator is `,`. We can fix this either by forcing the separator to always be `.` for decimals (and white space for thousands) - then change the `utils.getLocaleSeparators` function. Or use this function to get the separator.

Reply:
Cleaned up the comments, but sorting out the different local separators might take a bit to debug.

Reply:
you can change your OS settings, but you will also need to restart the browser after that

here is where it is on macos, on linux and windows it's also probably somewhere under language, time and numbers formatting
<img width="706" alt="Screenshot 2023-11-27 at 00 15 25" src="https://github.com/DeXter-on-Radix/website/assets/27793901/57b11c93-9a0d-4492-9377-bc184b3f8b49">

Reply:
if it causes a lot of problems you can either hardcode the `.` as a decimal separator and ` ` as thousands in `utils.getLocaleSeparators`, or just use whatever it returns now for `decimalSeparator` in your radix param (I vote for the second option, but in any case we need a single place of truth for the separator, let it be utils.ts, because why not)

Reply:
It is causing a bunch of problems and really causing some serious scope creep.

I'm thinking it isn't a blocker for launch, but a good quality of life in the future.

Maybe we can resolve this in another issue in the future.

Reply:
https://discord.com/channels/1125689933735145503/1125689934251032584/1179024092112883832